server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/aws/AwsS3AuthConfiguration.java

```diff
@@ -35,7 +38,15 @@
 
         @FunctionalInterface
         interface RequiredEndpoint {
-            RequiredAccessKeyId endpoint(String endpoint);
+            RequiredAccessKeyId endpoint(URI endpoint);
```

Comment:
It's cool but is it a breaking change? (some before valid and working configuration might now be rejected because they don't satisfy the URI format?)

Reply:
yes, before I accept an endpoint without protocol part. What should I do regrading this? update the `upgrade-instruction.md`?

Reply:
added